lib/Dialect/SPIRV/Serialization/Deserializer.cpp

```diff
@@ -1875,8 +1875,8 @@ LogicalResult ControlFlowStructurizer::structurizeImpl() {
       if (Block *mappedTo = mapper.lookupOrNull(newMerge))
         newMerge = mappedTo;
 
-      blockMergeInfo.try_emplace(newHeader, newMerge, newContinue);
       blockMergeInfo.erase(it);
+      blockMergeInfo.try_emplace(newHeader, newMerge, newContinue);
```

Comment:
Could you add a comment before `erase` to remind future changes?

" // The iterator should be erased before adding a new entry into blockMergeInfo to avoid iterator invalidation."

Reply:
thanks, fixed!